src/mrinufft/operators/off_resonance.py

```diff
@@ -0,0 +1,330 @@
+"""Off Resonance correction Operator wrapper.
+
+Based on the implementation of Guillaume Daval-Frérot in pysap-mri:
+https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/operators/fourier/orc_wrapper.py
+"""
+
+import math
+import numpy as np
+
+from .._utils import get_array_module
+
+from .base import FourierOperatorBase, CUPY_AVAILABLE, AUTOGRAD_AVAILABLE
+from .interfaces.utils import is_cuda_array
+
+if CUPY_AVAILABLE:
+    import cupy as cp
+
+if AUTOGRAD_AVAILABLE:
+    import torch
+
+
+def get_interpolators_from_fieldmap(
+    fieldmap, readout_time, n_time_segments=6, n_bins=(40, 10), mask=None
+):
+    r"""Create B and C matrices to approximate ``exp(-2j*pi*fieldmap*t)``.
+
+    Here, B has shape ``(n_time_segments, len(t))``
+    and C has shape ``(n_time_segments, *fieldmap.shape)``.
+
+    From Sigpy: https://github.com/mikgroup/sigpy
+    and MIRT (mri_exp_approx.m): https://web.eecs.umich.edu/~fessler/code/
+
+    Parameters
+    ----------
+    fieldmap : np.ndarray
+        Rate map defined as ``fieldmap = R2*_map + 1j * B0_map``.
+        ``*_map`` and ``t`` should have reciprocal units.
+        If ``zmap`` is real, assume ``zmap = B0_map``.
+        Also supports Cupy arrays and Torch tensors.
+        Expected shape is ``(nz, ny, nx)``.
+    readout_time : np.ndarray
+        Readout time in ``[s]`` of shape ``(nshots, npts)`` or ``(nshots * npts,)``.
+        Also supports Cupy arrays and Torch tensors.
+    n_time_segments : int, optional
+        Number of time segments. The default is ``6``.
+    n_bins : int | Sequence[int] optional
+        Number of histogram bins to use for ``(B0, T2*)``. The default is ``(40, 10)``
+        If it is a scalar, assume ``n_bins = (n_bins, 10)``.
+        For real fieldmap (B0 only), ``n_bins[1]`` is ignored.
+    mask : np.ndarray, optional
+        Boolean mask to avoid histogram of background values.
```

Comment:
Please make the description more user-oriented, like is it a mask of the background or the ROI ?